optaplanner-core/src/main/java/org/optaplanner/core/impl/statistic/SolverStatistic.java

```diff
@@ -0,0 +1,7 @@
+package org.optaplanner.core.impl.statistic;
+
+import org.optaplanner.core.api.solver.Solver;
+
+public interface SolverStatistic {
```

Comment:
This could be generic in `Solution_`, therefore eliminating some of the unchecked warnings you are suppressing.